rust/operator-binary/src/airflow_controller.rs

```diff
@@ -965,7 +915,25 @@ fn build_server_rolegroup_statefulset(
         };
         airflow_container.readiness_probe(probe.clone());
         airflow_container.liveness_probe(probe);
-        airflow_container.add_container_port("http", resolved_port.into());
+        airflow_container.add_container_port(HTTP_PORT_NAME, http_port.into());
+    }
+
+    if let Some(listener_class) =
+        airflow.merged_listener_class(airflow_role, &rolegroup_ref.role_group)
+    {
+        // all listeners will use ephemeral volumes as they can/should
+        // be removed when the pods are *terminated* (ephemeral PVCs will
+        // survive re-starts)
+        pb.add_listener_volume_by_listener_class(
+            LISTENER_VOLUME_NAME,
+            &listener_class.to_string(),
+            &recommended_labels,
+        )
+        .context(AddVolumeSnafu)?;
```

Comment:
Clients don't care about _which_ airflow webserver they connect to, right? In that case, we should use a group listener instead (https://docs.stackable.tech/home/stable/listener-operator/volume/#_listeners_stackable_techlistener_name), so we only end up with one load balancer that knows how to route everywhere.

Reply:
If I've understood this correctly, the operator would then be responsible for creating a listener-by-name (like we do for the kafka bootstrap listener) if the webserver role (or one of its rolegroups) declares a listener class as external-stable or external-unstable, and then this group listener is passed as the name to ListenerReference::ListenerClass for the ListenerOperatorVolumeSourceBuilder for the pvcs? And this listener doesn't care if webserver rolegroups specify different classes (e.g. one as external-stable and another as external-unstable)?

Reply:
Hmmm, but for Kafka we create a bootstrap-listener per role-group.

Reply:
0f650d3